lib/kennel/plans/printers/git.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require "open3"
+require "shellwords"
+require "tmpdir"
+
+module Kennel
+  module Plans
+    module Printers
+      class Git
+        DIFF_COMMAND = "(cd %s && git --no-pager diff --no-prefix --color=always --no-index %s %s)"
+
+        attr_reader :plan
+
+        def initialize(plan, io)
+          @plan = plan
+          @io = io
+        end
+
+        def print!
+          parent_directory, from_directory, to_directory = create_temporary_directories
```

Comment:
The whole business with temporary directories and file-writing can potentially be made much simpler by using temporary files or pipes.